holmes.go

```diff
@@ -65,68 +65,68 @@ func New(opts ...Option) (*Holmes, error) {
 
 // EnableThreadDump enables the goroutine dump.
 func (h *Holmes) EnableThreadDump() *Holmes {
-	h.opts.ThreadOpts.Enable = true
+	h.opts.threadOpts.Enable = true
 	return h
 }
 
 // DisableThreadDump disables the goroutine dump.
 func (h *Holmes) DisableThreadDump() *Holmes {
-	h.opts.ThreadOpts.Enable = false
+	h.opts.threadOpts.Enable = false
 	return h
 }
 
 // EnableGoroutineDump enables the goroutine dump.
 func (h *Holmes) EnableGoroutineDump() *Holmes {
-	h.opts.GrOpts.Enable = true
+	h.opts.grOpts.Enable = true
 	return h
 }
 
 // DisableGoroutineDump disables the goroutine dump.
 func (h *Holmes) DisableGoroutineDump() *Holmes {
-	h.opts.GrOpts.Enable = false
+	h.opts.grOpts.Enable = false
 	return h
 }
 
 // EnableCPUDump enables the CPU dump.
 func (h *Holmes) EnableCPUDump() *Holmes {
-	h.opts.CPUOpts.Enable = true
+	h.opts.cpuOpts.Enable = true
 	return h
 }
 
 // DisableCPUDump disables the CPU dump.
 func (h *Holmes) DisableCPUDump() *Holmes {
-	h.opts.CPUOpts.Enable = false
+	h.opts.cpuOpts.Enable = false
 	return h
 }
 
 // EnableMemDump enables the mem dump.
 func (h *Holmes) EnableMemDump() *Holmes {
-	h.opts.MemOpts.Enable = true
-	return h
-}
-
-// DisableMemDump disables the mem dump.
-func (h *Holmes) DisableMemDump() *Holmes {
-	h.opts.MemOpts.Enable = false
+	h.opts.memOpts.Enable = true
 	return h
 }
 
 // EnableGCHeapDump enables the GC heap dump.
 func (h *Holmes) EnableGCHeapDump() *Holmes {
-	h.opts.GCHeapOpts.Enable = true
+	h.opts.gCHeapOpts.Enable = true
 	h.finCh = make(chan struct{})
 	return h
 }
 
+// DisableMemDump disables the mem dump.
+func (h *Holmes) DisableMemDump() *Holmes {
+	h.opts.gCHeapOpts.Enable = false
+	return h
+}
+
 // DisableGCHeapDump disables the gc heap dump.
 func (h *Holmes) DisableGCHeapDump() *Holmes {
-	h.opts.GCHeapOpts.Enable = false
+	h.opts.gCHeapOpts.Enable = false
 	return h
 }
 
 func finalizerCallback(gc *gcHeapFinalizer) {
 	// disable or stop gc clean up normally
-	if !gc.h.opts.GCHeapOpts.Enable || atomic.LoadInt64(&gc.h.stopped) == 1 {
+	if !gc.h.opts.gCHeapOpts.Enable || atomic.LoadInt64(&gc.h.stopped) == 1 {
```

Comment:
```suggestion
	if atomic.LoadInt64(&gc.h.stopped) == 1 {
```

I think it's an old bug here.
we should not return when it's not enabled, otherwise, we can not enable it without `start`.